roles/httpd.lua

```diff
@@ -72,14 +72,32 @@ local function parse_listen(listen)
     return host, port, nil
 end
 
+-- get_params returns table with set params from config to
+-- pass it into new() func.
+local function get_params(params)
+    if params ~= nil then
+        return {
+            ssl_cert_file = params.ssl_cert_file,
+            ssl_key_file = params.ssl_key_file,
+            ssl_password = params.ssl_password,
+            ssl_password_file = params.ssl_password_file,
+            ssl_ca_file = params.ssl_ca_file,
+            ssl_ciphers = params.ssl_ciphers,
+        }
+    end
+end
```

Comment:
We need to validate the params in the role to avoid errors on the `apply` stage (and add unit-tests for the validation too).

Reply:
If http.server.new() does not have side effects than we could use it for validation in roles at the `validate` call.

Reply:
Used new function, added comment to it and unit tests on validation.